pages/flight-search.tsx

```diff
@@ -22,7 +24,37 @@ export default function FlightSearch(props: { flightSelectData: FlightSelectData
     router.push('/flight-select');
   };
 
-  const states = props.flightSelectData.data.states.map((state) => state.name);
+  // const states = props.flightSelectData.data.states.map((state) => state.name);
+
+  const fetcher = (url: URL, payload?: string) => {
+    const options = {
+      method: payload ? 'POST' : 'GET',
+      ...(payload && { body: payload }),
+      headers: {
+        accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    };
+
+    return fetch(url, options).then((res) => res.json());
+  };
+
+  const bodyContent = { country: 'United States' };
+
+  const { data, error } = useSWR<FlightSelectData>(
+    ['https://countriesnow.space/api/v0.1/countries/states', JSON.stringify(bodyContent)],
+    fetcher,
+  );
+
+  useEffect(() => {
+    if (data) {
+      setTimeout(() => {
+        setLoading(!data && !error);
```

Comment:
is the set timeout so it doesn't flash and correctly displays the loader?

Reply:
Sorry, I should've explained that ahead of time.  We wouldn't use setTimeout at all.  I just did it to emulate a slower api call and to be able to see it all happening on the page.  Otherwise it happened too fast to be able to see.